### DISCUSSION TITLE
play() request was interrupted?

### DISCUSSION THREAD

**Discussion by jubejubess** [jspsych/jsPsych#1169] [Q&A]
Hello!

I was stress-testing a new experiment and I seemed to run into some minor issues. The audio and video playback is perfect for the most part, but on rare occasions, the audio and/or video does not play at all. I checked and either one of the two errors show up in the console when this occurs: 

```
Uncaught (in promise) DOMException: The play() request was interrupted by a call to pause().
Uncaught (in promise) DOMException: The play() request was interrupted by a new load request.
```

I was wondering what causes this issue and if there was a work-around to this? 

Thanks for any insight!

**Comment by becky-gilbert** [comment]
[This page](https://developers.google.com/web/updates/2017/06/play-request-was-interrupted) explains the problem. It sounds like in some cases there are delays in your audio and/or video loading/playing, and then the plugin calls the .pause() function before the audio/video has started playing, which interrupts the loading that was initiated by .play(). The suggested fix for this is to use the Promise returned by the .play() function to see if playback was successful, and use that to trigger everything that should happen after that, including .pause(). One potential problem with that fix is that it may not work in all browsers, for instance it looks like the .play() function doesn't return a promise in Edge (see [here](https://developers.google.com/web/updates/2017/06/play-request-was-interrupted)). 

If you don't want to use promises, then another way to do this is create an event handler that fires when the media starts playing, and use that function to trigger the rest of the trial. But keep in mind that you'll also need to handle the case where that event never fires (because the media doesn't load/play for some reason), for instance by ending the trial if loading problems are detected or after some fixed duration. The media element properties, events and methods listed [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement) might be helpful if you want to go that route.

I was also going to suggest that you make sure you've done everything you can to prevent loading audio/video delays during the experiment, but I see that I've already told you everything I know about preloading etc. in [this thread](https://github.com/jspsych/jsPsych/discussions/1066#discussioncomment-87255) 😄

**Reply by isabelletelford** [reply]
@jubejubess please let me know if you found a good fix to this! I'm having a similar problem but it's happening for playback of certain audio files (but not all) every time I run the experiment.
